runtime/resolvers/metricsview_sql.go

```diff
@@ -0,0 +1,257 @@
+	}, nil
+}
+
+// expandMetricsViewSQL parses the metrics SQL where it
+//
+// 1. expands AGGREGATE(metric) into actual aggregate definition from model_view definition
+// 2. converts the FROM model_view clause to the underlying FROM table
+//
+// example transformation:
+// input : SELECT dim_col, AGGREGATE(metric) FROM metrics_view GROUP BY dim_col
+// output : SELECT dim_name, sum(col2) FROM metrics_view_model  GROUP BY dim_col
+func expandMetricsViewSQL(ctx context.Context, ctrl *runtime.Controller, opts *runtime.APIResolverOptions, sql string) (string, []*runtimev1.ResourceName, error) {
+	// 1. get the dependencies
+	meta, err := compilerv1.AnalyzeTemplate(sql)
+	if err != nil {
+		return "", nil, err
+	}
+	var deps []*runtimev1.ResourceName
+	for _, ref := range meta.Refs {
+		deps = append(deps, &runtimev1.ResourceName{Kind: ref.Kind.String(), Name: ref.Name})
+	}
+
+	// 2. Expand from metrics_view
+	// if there is a match, it will be of the form `from (metrics_view)``
+	// first is full match second is `metrics_view`
+	matches := fromMVRegex.FindAllStringSubmatch(sql, -1)
+
+	mvToMeasureExprMap := make(map[string]map[string]string)
+	var mvConnector string
+	for _, match := range matches {
+		metricView := unquote(match[1])
+		if _, ok := mvToMeasureExprMap[metricView]; ok {
+			continue
+		}
+
+		resource, err := ctrl.Get(ctx, &runtimev1.ResourceName{Kind: runtime.ResourceKindMetricsView, Name: metricView}, false)
+		if err != nil {
+			if errors.Is(err, drivers.ErrResourceNotFound) {
+				continue
+			}
+			return "", nil, fmt.Errorf("error fetching resource %v: %w", metricView, err)
+		}
+		if resource.GetMetricsView() == nil { // resource is not a metrics view
+			continue
+		}
+		deps = append(deps, &runtimev1.ResourceName{Kind: runtime.ResourceKindMetricsView, Name: metricView})
+
+		if mvConnector != "" && resource.GetMetricsView().Spec.Connector != mvConnector {
+			return "", nil, fmt.Errorf("all referenced metrics views must use the same connector")
+		}
+		mvConnector = resource.GetMetricsView().Spec.Connector
+
+		// change from metric view to underlying table
+		fromQry, measureToExprMap, err := underlyingTableQuery(ctrl.Runtime, opts, resource.GetMetricsView(), resource.Meta.StateUpdatedOn.AsTime())
+		if err != nil {
+			return "", nil, err
+		}
+		mvToMeasureExprMap[metricView] = measureToExprMap
+		sql = strings.ReplaceAll(sql, match[0], fromQry)
+	}
+
+	// 3. Expand aggregate expression
+	if len(mvToMeasureExprMap) > 0 {
+		// example query = select dim1, aggregate(mv."my measure") from mv
+		// captures AGGREGATE("mv name"."my measure"), my name, measure
+		aggMatches := aggregateRegex.FindAllStringSubmatch(sql, -1)
+		for _, aggMatch := range aggMatches {
+			metricView := unquote(aggMatch[1])
+			var expr string
+			var found bool
+			if metricView == "" {
+				if len(mvToMeasureExprMap) > 1 {
+					return "", nil, fmt.Errorf("measure should be precedded with metric_view if api references more than one metric view")
```

Comment:
nit: spelling of `precedded`. Also would suggest a more explicit error message, something like `ambiguous reference to measure %q: use a fully qualified name such as "metrics_view.measure"`